Add Readme

Tieqiong opened the pull request:
Fix #12 

How it [looks like](https://github.com/Tieqiong/readme/tree/0296420fdd0f22d43d8545e1d2adf275b83e4e8e) after cookiecut

Need discussion and changes:

- project icon
-- Higher resolution
-- Make it the GitHub org avatar?
- Badges
-- DOI
-- Test
- Contributor.rst in diffpy.utils?

sbillinge reviewed (commented):
wow, this looks great.  Our code just got better!  Thanks for this @Tieqiong !

Here are few thoughts.

1. we probably need a higher resolution image of the diffpy icon (though we don;t want a higher res version as the github avatar I am guessing).  Just need to think about this.
2. add a test seems to be pointing to matplotlib GH workflow. Not sure if this adds much value?  We could either cut this or point to instructions how to add at test?
3. we don't really want to offer people to get their own doi, but we do want to have them cite the paper that describes the package (if it exists).  Can we have something like "Cite this package" and link to the paper describing it?  By default it could link to the current GH website, or maybe to the release page?  Later we can add text with the correct citation and a link to the actual paper?
4. The installation integrity seems to have an extra "pip install" line. regardless of how the code was installed the `tests.run` should work.  Also, does the user have to be in a particular directory for that to work?  If so, make that explicit or lots of people will have tests that fail and reach out to us but they were just running it from the wrong directory.
5. It might be a good place, right after this "test", a section of "Reporting Bugs, issues and requesting features".   The top thing there I would like to be to ask them to join the diffpy-users community by joining the Google group.  The second thing would be to report a bug as an issue and/or submit a fix to a bug as a PR and link to thecontributing page.
6. I doubt we want the ADD MORE and ANYTHING ELSE in the Development section.  Can we just agree on text that encourages the community of scientists to get involved, but point them to our code of conduct and contributing instructions?  It is good practice to think about making these as inclusive as possible.  We can again take inspiration from Matplotlib and other successful libraries.  Put edits to these file on different issues/PRs but link to them here.
7. most packages won't have their own web-page ,but should be available through diffpy.org.  Either link to diffpy.org here or better.

Super!

Tieqiong commented:
@sbillinge Thanks for the comments!

1. I'm thinking of changing the avatar of the diffpy GitHub account from the default to the diffpy icon. This is beneficial because 1. It makes the cookie cutter easier to fit for both diffpy and non-diffpy packages (if we are going to have a different icon for a non-diffpy package) and 2. make the diffpy account looks cool.
2. I linked the Test badge to Matplotlib GH workflow (test.yml to be more precise) because we were talking about adding a test badge yesterday and Matplotlib created this badge (not a lot of packages have this test badge) by having the test.yml in their workflow. It might be not worth it to add another file in our workflow so I'm thinking of other ways.
3. Yes instead of having individual and separate DOI for each release of the packages (like what scikit learn does) many other packages (like Numpy and Scipy) link to the paper. I also think having the paper DOI placed here is a great idea.
4. Yes I'll check and add more precise instructions on testing.
5. Sounds good, will add a new section "Reporting Bugs, issues and requesting features".
6. I'll think about text that can encourages the community to get involved. One problem is diffpy.utils doesn't have a separate Contribution page but I assume we have one in the cookiecutter template.
7. I'll make diffpy.org to be the web link here. Do we want to consider the case for non-diffpy packages?

sbillinge commented:
> @sbillinge Thanks for the comments!
> 
>     1. I'm thinking of changing the avatar of the diffpy GitHub account from the default to the diffpy icon. This is beneficial because 1. It makes the cookie cutter easier to fit for both diffpy and non-diffpy packages (if we are going to have a different icon for a non-diffpy package) and 2. make the diffpy account looks cool.
> 
>     2. I linked the Test badge to Matplotlib GH workflow (test.yml to be more precise) because we were talking about adding a test badge yesterday and Matplotlib created this badge (not a lot of packages have this test badge) by having the test.yml in their workflow. It might be not worth it to add another file in our workflow so I'm thinking of other ways.
> 
>     3. Yes instead of having individual and separate DOI for each release of the packages (like what scikit learn does) many other packages (like Numpy and Scipy) link to the paper. I also think having the paper DOI placed here is a great idea.
> 
>     4. Yes I'll check and add more precise instructions on testing.
> 
>     5. Sounds good, will add a new section "Reporting Bugs, issues and requesting features".
> 
>     6. I'll think about text that can encourages the community to get involved. One problem is diffpy.utils doesn't have a separate Contribution page but I assume we have one in the cookiecutter template.
> 
>     7. I'll make diffpy.org to be the web link here. Do we want to consider the case for non-diffpy packages?

sounds good.  Make the changes and we can review.  For 7, it is a good point.  I am a bit reluctant to add more questions, but we could add a "link to diffpy.org?" question and have default y but allow n.  If n is selected, just put nothing there as the documentation will basically be the README itself (and any github.io docs)

Tieqiong commented:
[Preview](https://github.com/Tieqiong/readme/blob/3cb2ec223403b6e48db6750a8d9869f1803d0958/README.rst)

@sbillinge things left: 
1. We don't need to add a "link to diffpy.org" question anymore (thanks to @Sparks29032), but a doi question could be helpful for both the doi badge and citation link
2. I added the google user groups. What about the non-diffpy packages?
3. I linked to a code of conduct part (as suggested) so we need an extra CODE_OF_COUNDUCT.rst
4. I modified the install section a little bit. The `pip install` in check integrity is to install the package requirement for testing (namely pytest-mock and freezegun). I believe it's necessary or there'll be errors

sbillinge commented:
> [Preview](https://github.com/Tieqiong/readme/blob/3cb2ec223403b6e48db6750a8d9869f1803d0958/README.rst)
> 
> @sbillinge things left:
> 
>     1. We don't need to add a "link to diffpy.org" question anymore (thanks to @Sparks29032), but a doi question could be helpful for both the doi badge and citation link
> 
>     2. I added the google user groups. What about the non-diffpy packages?
> 
>     3. I linked to a code of conduct part (as suggested) so we need an extra CODE_OF_COUNDUCT.rst (see [Add code of conduct #83](https://github.com/Billingegroup/cookiecutter/pull/83))
> 
>     4. I modified the install section a little bit. The `pip install` in check integrity is to install the package requirement for testing (namely pytest-mock and freezegun). I believe it's necessary or there'll be errors
> 
>     5. Now I'm using the Github avatar as the project icon, so we need to change the diffpy organization avatar to the diffpy icon to see the right img

1. we don't need a DOI question or a doi badge.  We can hard-code text as a place-holder for a citation link that will be filled in later.  No question needed.  We won't have this info at the time of creating the repo.
2. For the user group we don't use diffpy-dev.  Please just have them join diffpy-users.  They can post bugs there too (they do anyway).  We just use that one place for all the group code, so we can use the same user-group also for non-diffpy packages.
3. I saw the code of conduct.  I left a comment, but basically, it looks great.
4. In that case, let's remove that.  We don't want users installing those things in their envs.    We can just remove that text.    It is mostly only needed for apps anyway, not packages per se (do you ever do that for numpy for example?)
5. ok, I can do that.

Tieqiong commented:
[Preview](https://github.com/Tieqiong/readme/blob/3a87cd5e44e6b29228a466b3daeb33d3832f6f15/README.rst)

Tieqiong commented:
[Preview](https://github.com/Tieqiong/readme/blob/8eb8b5a8a5e5f867e4a63304706a14b58bd37b74/README.rst)

Sparks29032 commented on {{ cookiecutter.repo_name }}/README.rst:
```diff
@@ -1,22 +1,104 @@
+
+* LONGER DESCRIPTION HERE
+
+For more information about the {{ cookiecutter.project_name }} library, see the `users manual <https://{{ cookiecutter.github_org }}.github.io/{{ cookiecutter.project_name }}>`_.
```
Change "see the `users manual`" to "consult our `documentation`"

Sparks29032 commented on {{ cookiecutter.repo_name }}/README.rst:
```diff
@@ -1,22 +1,104 @@
+We want to install our packages in a suitable conda environment.
+The following creates and activates a new environment named ``{{ cookiecutter.project_name }}`` ::
+
+        conda create -n {{ cookiecutter.project_name }} python=3
+        conda activate {{ cookiecutter.project_name }}
+
+Then, to fully install ``{{ cookiecutter.project_name }}`` in our active environment, run ::
+
+        conda install {{ cookiecutter.project_name }}
```
Change all references from `project_name` to `package_dist_name` in the installation section
